app.py

```diff
@@ -0,0 +1,81 @@
+from flask import Flask, render_template, request, redirect, url_for, flash, session
+from flask_sqlalchemy import SQLAlchemy
+from flask_bcrypt import Bcrypt
+from config import Config
+from models import db, User
+
+app = Flask(__name__)
+app.config.from_object(Config)
+
+# Initialize extensions
+db.init_app(app)
+bcrypt = Bcrypt(app)
+
+# Create tables explicitly during app initialization
+with app.app_context():
+    db.create_all()
+
+# Routes
+@app.route('/')
+def home():
+    return render_template('index.html')
+
+@app.route('/register', methods=['GET', 'POST'])
+def register():
```

Comment:
As per `Single Responsibilities Principle` each method should perform exactly one job. But in your case, this method is perfroming following operations:

1. Checking its method type
2. Validation check with db
3. Insertion data to db

**Also, I can't see any data validation for submitted request.** We need to ensure, no malicious things should be inserted to db and we should have some username and email string size constraint compatible with our db schema.

Reply:
You can utilize `Repository Pattern`. Its just for your reference. But at least make a separate function that will insert data to db.